tracer/src/Datadog.Trace/Configuration/ExporterSettings.cs

```diff
@@ -66,37 +59,50 @@ public ExporterSettings()
         /// </summary>
         /// <param name="source">The <see cref="IConfigurationSource"/> to use when retrieving configuration values.</param>
         public ExporterSettings(IConfigurationSource source)
-            : this(source, File.Exists)
         {
-        }
+            ValidationWarnings = new List<string>();
+            // The settings here should not do further checks that the setter of the API does (thus only the type should be checked basically)
+            // All the logic to assign the transport will be in the ImmutableExporterSettings .ctor
 
-        /// <summary>
-        /// Initializes a new instance of the <see cref="ExporterSettings"/> class.
-        /// Direct use in tests only.
-        /// </summary>
-        internal ExporterSettings(IConfigurationSource source, Func<string, bool> fileExists)
-        {
-            _fileExists = fileExists;
+            var traceAgentUrl = source?.GetString(ConfigurationKeys.AgentUri);
+            if (!string.IsNullOrWhiteSpace(traceAgentUrl))
+            {
+                if (!Uri.TryCreate(traceAgentUrl, UriKind.RelativeOrAbsolute, out var uri))
+                {
+                    ValidationWarnings.Add($"The provided Uri: ${traceAgentUrl} provided in '{ConfigurationKeys.AgentUri}' is not valid. It won't be taken into account to send traces.");
+                }
+                else
+                {
+                    AgentUri = uri;
+                }
+            }
 
-            ConfigureTraceTransport(source, out var shouldUseUdpForMetrics);
-            ConfigureMetricsTransport(source, shouldUseUdpForMetrics);
+            TracesUnixDomainSocketPath = source?.GetString(ConfigurationKeys.TracesUnixDomainSocketPath);
```

Comment:
I just noticed while looking at the documentation PR (https://github.com/DataDog/documentation/pull/13131) that this setting maps to `DD_APM_RECEIVER_SOCKET`. That is an Agent setting and shouldn't be a Tracer settings. I don't see anything about using `DD_APM_RECEIVER_SOCKET` in the RFC, it doesn't follow the `DD_TRACE_*` convention, and other Tracers seem to only use `DD_TRACE_AGENT_URL` to configure Unix Domain Sockets.

If it makes the logic here easier, we should consider removing this setting. Otherwise, we should at least rename it (and I guess keep supporting the older name if anyone is already using it).

Reply:
Fixed in e445c9959f19dcc8ec0607a401153d98f9803479